src/uct/base/uct_iface.c

```diff
@@ -264,19 +264,18 @@ int uct_iface_is_reachable_params_valid(
     return 1;
 }
 
-int uct_base_iface_is_reachable_v2(
-        const uct_iface_h iface, const uct_iface_is_reachable_params_t *params)
+int uct_iface_is_reachable_params(
+        const uct_iface_h iface, const uct_iface_is_reachable_params_t *params,
+        uint64_t flags, uct_iface_is_reachable_v2_func_t tl_iface_is_reachable)
```

Comment:
IMO using function pointer arg here is overcomplicating

Reply:
looks quite clear though

Reply:
fixed by using three steps checks (params, tl reach, scope reach) in tl specific functions